### DISCUSSION TITLE
Ceiling Fan Restarting several times

### DISCUSSION THREAD

**Discussion by heffneil** [arendst/Tasmota#20928] [Q&A]
### PROBLEM DESCRIPTION
I have a Single Pole Martin Jerry Switch it is running Tasmota 12.4.  I have other switches (different brands) that control ceiling fans and we have a weird behavior where the fan will turn off and restart.   It appears to do this 3 times, in one example.  This seems to happen different times once a day (That we observe).  It happened last night but the switch / tasmota reflect:

Uptime | 61T11:53:24

so I don't belive the switch is restarting.  The question is would a disconnect from the wifi network cause the power to the load to change?

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [ ] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [ ] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [ ] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [ ] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [ ] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [ ] Device used (e.g., Sonoff Basic): _____
- [ ] Tasmota binary firmware version number used: _____
  - [ ] Pre-compiled
  - [ ] Self-compiled
- [ ] Flashing tools used: _____
- [ ] Provide the output of command: `Backlog Template; Module; GPIO 255`:
```lua
  Configuration output here:

09:44:04.746 MQT: stat/guestroom-fan/RESULT = {"NAME":"MJ-S01 2Way Switch","GPIO":[1,1,1,1,288,321,0,0,224,32,544,1,0,0],"FLAG":0,"BASE":18}
09:44:04.971 MQT: stat/guestroom-fan/RESULT = {"Module":{"0":"MJ-S01 2Way Switch"}}
09:44:05.227 MQT: stat/guestroom-fan/RESULT = {"GPIO0":{"0":"None"},"GPIO1":{"0":"None"},"GPIO2":{"0":"None"},"GPIO3":{"0":"None"},"GPIO4":{"288":"Led1"},"GPIO5":{"321":"Led_i2"},"GPIO9":{"0":"None"},"GPIO10":{"0":"None"},"GPIO12":{"224":"Relay1"},"GPIO13":{"32":"Button1"},"GPIO14":{"544":"LedLink"},"GPIO15":{"0":"None"},"GPIO16":{"0":"None"},"GPIO17":{"0":"None"}}

```
- [ ] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:
```lua
  Rules output here:

09:43:34.481 MQT: stat/guestroom-fan/RESULT = {"Rule1":{"State":"OFF","Once":"OFF","StopOnError":"OFF","Length":144,"Free":367,"Rules":"on power1#state do backlog ledpower1 %value%; ledpower2 %value% endon       on power1#boot do backlog ledpower1 %value%; ledpower2 %value% endon"}}
09:43:34.707 MQT: stat/guestroom-fan/RESULT = {"Rule2":{"State":"OFF","Once":"OFF","StopOnError":"OFF","Length":0,"Free":511,"Rules":""}}
09:43:34.957 MQT: stat/guestroom-fan/RESULT = {"Rule3":{"State":"OFF","Once":"OFF","StopOnError":"OFF","Length":0,"Free":511,"Rules":""}}
```
- [ ] Provide the output of this command: `Status 0`:
```lua
  STATUS 0 output here:

09:44:33.116 MQT: stat/guestroom-fan/STATUS = {"Status":{"Module":0,"DeviceName":"Guestroom  Fan","FriendlyName":["Guestroom Fan"],"Topic":"guestroom-fan","ButtonTopic":"0","Power":1,"PowerOnState":3,"LedState":8,"LedMask":"FFFF","SaveData":1,"SaveState":1,"SwitchTopic":"0","SwitchMode":[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],"ButtonRetain":0,"SwitchRetain":0,"SensorRetain":0,"PowerRetain":1,"InfoRetain":0,"StateRetain":0,"StatusRetain":0}}
09:44:33.124 MQT: stat/guestroom-fan/STATUS1 = {"StatusPRM":{"Baudrate":115200,"SerialConfig":"8N1","GroupTopic":"tasmotas","OtaUrl":"http://ota.tasmota.com/tasmota/release/tasmota.bin.gz","RestartReason":"Software/System restart","Uptime":"61T12:00:11","StartupUTC":"2024-01-10T01:44:22","Sleep":50,"CfgHolder":4617,"BootCount":79,"BCResetTime":"2023-03-26T15:00:17","SaveCount":543,"SaveAddress":"F9000"}}
09:44:33.129 MQT: stat/guestroom-fan/STATUS2 = {"StatusFWR":{"Version":"12.4.0(tasmota)","BuildDateTime":"2023-02-24T15:32:38","Boot":31,"Core":"2_7_4_9","SDK":"2.2.2-dev(38a443e)","CpuFrequency":80,"Hardware":"ESP8266EX","CR":"404/699"}}
09:44:33.133 MQT: stat/guestroom-fan/STATUS3 = {"StatusLOG":{"SerialLog":2,"WebLog":2,"MqttLog":0,"SysLog":0,"LogHost":"","LogPort":514,"SSId":["IOTDevices",""],"TelePeriod":300,"Resolution":"558180C0","SetOption":["4022A129","2805C80001000600003C5A0A002800000000","00008268","00006000","00004000","00000000"]}}
09:44:33.146 MQT: stat/guestroom-fan/STATUS4 = {"StatusMEM":{"ProgramSize":632,"Free":368,"Heap":18,"ProgramFlashSize":1024,"FlashSize":2048,"FlashChipId":"1540C8","FlashFrequency":40,"FlashMode":"DOUT","Features":["00000809","8F9AC787","04368001","000000CF","010013C0","C000F981","00004004","00001000","54000020","00000080"],"Drivers":"1,2,3,4,5,6,7,8,9,10,12,16,18,19,20,21,22,24,26,27,29,30,35,37,45,62","Sensors":"1,2,3,4,5,6"}}
09:44:33.154 MQT: stat/guestroom-fan/STATUS5 = {"StatusNET":{"Hostname":"guestroom-fan","IPAddress":"172.16.1.149","Gateway":"172.16.0.1","Subnetmask":"255.255.252.0","DNSServer1":"192.168.10.99","DNSServer2":"192.16.10.251","Mac":"98:F4:AB:BD:FB:6D","Webserver":2,"HTTP_API":1,"WifiConfig":4,"WifiPower":17.0}}
09:44:33.159 MQT: stat/guestroom-fan/STATUS6 = {"StatusMQT":{"MqttHost":"mqtt.home","MqttPort":1883,"MqttClientMask":"DVES_%06X","MqttClient":"DVES_BDFB6D","MqttUser":"mqtt-user","MqttCount":82,"MAX_PACKET_SIZE":1200,"KEEPALIVE":30,"SOCKET_TIMEOUT":4}}
09:44:33.165 MQT: stat/guestroom-fan/STATUS7 = {"StatusTIM":{"UTC":"2024-03-11T13:44:33","Local":"2024-03-11T09:44:33","StartDST":"2024-03-10T02:00:00","EndDST":"2024-11-03T02:00:00","Timezone":99,"Sunrise":"08:19","Sunset":"20:10"}}
09:44:33.171 MQT: stat/guestroom-fan/STATUS10 = {"StatusSNS":{"Time":"2024-03-11T09:44:33"}}
09:44:33.176 MQT: stat/guestroom-fan/STATUS11 = {"StatusSTS":{"Time":"2024-03-11T09:44:33","Uptime":"61T12:00:11","UptimeSec":5313611,"Heap":18,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":82,"POWER":"ON","Wifi":{"AP":1,"SSId":"IOTDevices","BSSId":"C8:B5:AD:30:BE:A1","Channel":1,"Mode":"11n","RSSI":74,"Signal":-63,"LinkCount":17,"Downtime":"0T01:20:09"}}}
```
- [ ] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:
```lua
  Console output here:


```

### TO REPRODUCE
_Steps to reproduce the behavior:_

Not really a way to reproduce it happens randomly during the course of the day.

### EXPECTED BEHAVIOUR
_A clear and concise description of what you expected to happen._

Regardless of switch network status the load should remain on.

### SCREENSHOTS
_If applicable, add screenshots to help explain your problem._

### ADDITIONAL CONTEXT
_Add any other context about the problem here._

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by Jason2866** [comment]
The device has reconnects from mqtt (82) and WiFi (17). There is no restart (boots) since 61 days.
Device was offline for 01h20.
`MqttCount":82,"POWER":"ON","Wifi":{"AP":1,"SSId":"IOTDevices","BSSId":"C8:B5:AD:30:BE:A1","Channel":1,"Mode":"11n","RSSI":74,"Signal":-63,"LinkCount":17,"Downtime":"0T01:20:09"}}}`

What the reason for this is? No clue. Maybe wifi network congestion? I see nothing what can be done from Tasmota side.

I have not seen any power state change with all my Tasmota devices when wifi goes off/online.

**Comment by heffneil** [comment]
What does offline mean? It means it wasn't connected to the mqtt server?  If that is the case it should not change the state of the load (power) to the device, correct?  Would a reconnect to the MQTT server cause the state to change?  Trying to get to the bottom of this?

**Reply by Jason2866** [reply]
Check the logs of the mqtt broker if you see changes of the power state. Offline does mean Tasmota Device had no wifi connection. A reconnect to the mqtt broker normally does not change power state.

**Reply by sfromis** [reply]
The mentioned "Downtime" is under the Wifi object in the JSON, hence that's what it means. MqttCount being much higher than WifiCount suggests some significant instability around MQTT, without it only being when Wifi is also not with good availability.

**Comment by heffneil** [comment]
I guess the question is there isn't a setting (wifi config?) that would cause the ESP in this switch to reconnect this often?  Obviously the switch and the AP aren't moving so the cause for wifi instability means other problems. 

I just want to make sure that the MQTT and or Wifi reconnect wouldn't cause the switch to turn off and then back on seemingly immediately.

Any suggestions on how to log the turning on and off of the switch when it occurs? Thanks!

**Reply by sfromis** [reply]
When it comes to actions performed, the usual answer is "it depends". Like if you have lingering retained commands being picked up after every MQTT reconnect, or rules doing things intended to be after boot.

When it is about not having stable Wifi, options for following what happens in between gets limited, unless you can have a serial connection. If you can get in quick after a connection outage, the console should still show a limited amount of messages back in time.

Outside of considerations around network outages, you can also have logs sent to a syslog server on your network, or use `MqttLog` to do logging via the MQTT broker.

**Comment by heffneil** [comment]
I don't have any actions performed, you can see the rules aren't even enabled on this switch.  I don't have any other system controlling this particular load so I'm not sure what it could be depending on.

**Reply by sfromis** [reply]
You are yourself asking about actions performed, as in "wouldn't cause the switch to turn off and then back on seemingly immediately". If you have retained MQTT commands, they will kick in at that time, and could be a thing you're dealing with.

And you clearly have symptoms of bad instability around both Wifi and MQTT.

**Comment by heffneil** [comment]
How often is the mqtt count refreshed?

**Reply by sfromis** [reply]
When it reconnects after some type of outage.

**Comment by heffneil** [comment]
I mean it shows 82 times.  Is that 82 times in the life of the device?  Is that this week? 24 hour period?  When does it go to 0?

**Reply by sfromis** [reply]
Since last boot.
